### DISCUSSION TITLE
Layer weights initialization

### DISCUSSION THREAD

**Discussion by kimzed** [pyg-team/pytorch_geometric#4650] [Q&A] [answered]
I am trying to debug my model which is not working properly, and I asked myself a question regarding weight initialization.

Usually in other architecture (e.g. CNN model) we initialize the layers weight in the model class. I do not find that in any tutorial about GNNs, so I was wondering where does this take place? Am i supposed to do the initialization somewhere? I suppose it is done automatically but i would like to be sure about it.

**Comment by rusty1s** [comment, marked as answer]
Yes, initialization is done automatically as in regular PyTorch. You can customize them by overriding `reset_parameters()` or following [this thread](https://stackoverflow.com/questions/51125856/custom-weight-initialization-in-pytorch).

**Reply by kimzed** [reply]
and what is the standard initialization method? is it common practice to use the standard one?

**Reply by rusty1s** [reply]
Yes, I think it is common practice. For some operators, e.g., GCN, we make use of the initialization technique proposed by the authors.
